tools/extract_metadata.py

```diff
@@ -308,6 +304,8 @@ def extract_metadata(filename):
           em_js_func_types[i.field] = types[i.type]
         else:
           import_names.append(i.field)
+      elif i.kind in (webassembly.ExternType.GLOBAL, webassembly.ExternType.TAG):
+        import_names.append(i.field)
```

Comment:
Can this code be deduplicated with the similar code above on lines 242-249?

Reply:
I couldn't come up with a particularly nice way to remove all the duplication here.. I did remove some of it.